### ISSUE TITLE
Document the process for using translate.zanata.org

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#297]
When new text is added to the site, such as with "Legend updates" issue #282, it needs to be captured in both Hebrew and English. Currently, the knowledge of how to create new text place holders, upload them to [translate.zanata.org's IsraelHiking project](https://translate.zanata.org/project/view/IsraelHiking), download the translations back, and update the site code, is undocumented.

**Comment by HarelM** [comment]
@zstadler please let me know if the process I wrote about is clear enough.
https://github.com/IsraelHikingMap/Site/wiki/Update-Translations
I created our first wiki pages, I think this is the right place for this sort of documentation, but I wasn't sure.
Please close this issue if so, feel free to change the instructions to be more understandable by you or the next person to read it.

**Comment by zstadler** [comment]
I'm getting an error opening Visual Studio's Task Runner Explorer (Ctrl+Alt+Bkspce):
```
Failed to run "...\GitHub\IsraelHikingMap\Site\IsrealHiking.Web\Gulpfile.js"...
cmd.exe /c gulp --tasks-simple
...\GitHub\IsraelHikingMap\Site\IsrealHiking.Web\node_modules\gulp-filter\index.js:2
const path = require('path');
^^^^^
SyntaxError: Use of const in strict mode.
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (...\GitHub\IsraelHikingMap\Site\IsrealHiking.Web\gulpfile.js:9:12)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
```

**Comment by zstadler** [comment]
It may be worthwhile to [install the Zanata CLI tool](http://docs.zanata.org/en/release/client/installation/windows-installation/) and automate the [uploading](http://docs.zanata.org/en/release/client/commands/push/) and [downloading](http://docs.zanata.org/en/release/client/commands/pull/) tasks.

**Comment by HarelM** [comment]
Open task runner:
![image](https://cloud.githubusercontent.com/assets/3269297/22400465/516a595e-e5be-11e6-9c1a-b27ca5a08167.png)

Run Task:
![image](https://cloud.githubusercontent.com/assets/3269297/22400479/a6021178-e5be-11e6-8257-9e9502bd1004.png)

I might have configured things on my PC that are not configured in yours, I would very much like to know if this is the case and update the setup instructions.

Are the images above close to what you see in Visual Studio?

**Comment by HarelM** [comment]
> It may be worthwhile to install the Zanata CLI tool and automate the uploading and downloading tasks.
Maybe if it was wrapped in an npm package I could add to the gulp task, and even so there will be a password and username issue there that I won't be able to check-in to github due to publicity.
bottom line: It saves 3 clicks, which I don't mind doing since I'm going to do a lot more while I translate the text... feel free to explore it though :-)

**Comment by zstadler** [comment]
The error I'm getting is the same as sindresorhus/gulp-imagemin#178
The isssue was resolved by installing and using node with a higher version than the default in Visual Studio 2015.

Have you done that? If so, the instructions for [setting-up this project](/IsraelHikingMap/Site#setting-up-this-project) need to be updated.

**Comment by zstadler** [comment]
Also, see [this stackoverflow answer](http://stackoverflow.com/a/36790307)

**Comment by HarelM** [comment]
Please update the setting-up the project with this new info according to the steps you made to make it work. I probably did it while playing around to understand the whole Node.js environment. Since this is globally installed on the computer regardless of Visual Studio it probably affected it. 
I'm sorry you encounter all these issues, but it's a good thing :-).

**Comment by HarelM** [comment]
@zstadler is this issue still open? Is there something I can do in order to close it?

**Comment by HarelM** [comment]
I went over the [Wiki page](https://github.com/IsraelHikingMap/Site/wiki/Adding-New-Text-and-Updating-the-Translations) it looks great!
I'm closing this issue as I was able to make things work on a clean PC using the instruction I wrote.
We might want to reduce some text on the main page by moving some of the stuff to the Wiki, for example the setting up the project etc. We can link from the main page to relevant pages in the wiki IMO.
